stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -114,23 +138,74 @@ class PaymentSheet internal constructor(
         }
     }
 
+    /**
+     * A class that presents the individual steps of a payment sheet flow.
+     */
     internal interface FlowController {
-        fun getPaymentOption(): PaymentOption?
 
-        fun configure(
+        /**
+         * Configure the FlowController to process a [PaymentIntent].
+         *
+         * @param paymentIntentClientSecret the client secret for the [PaymentIntent].
+         * @param configuration optional [PaymentSheet] settings.
+         * @param callback called with the result of configuring the FlowController.
+         */
+        fun configureWithPaymentIntent(
             paymentIntentClientSecret: String,
-            configuration: Configuration,
+            configuration: Configuration?,
             callback: ConfigCallback
         )
 
-        fun configure(
+        /**
+         * @see [configureWithPaymentIntent] above.
+         */
+        fun configureWithPaymentIntent(
             paymentIntentClientSecret: String,
             callback: ConfigCallback
+        ) {
+            configureWithPaymentIntent(paymentIntentClientSecret, null, callback)
+        }
```

Comment:
If we have a method that doesn't take a `Configuration`, should the method that does take a `Configuration` require it to be not-null? This might be clearer for users.

Reply:
Removed overloads.